packages/core/src/PartialExecutionUtils/getIncomingData.ts

```diff
@@ -7,18 +6,8 @@ export function getIncomingData(
 	runIndex: number,
 	connectionType: NodeConnectionType,
 	outputIndex: number,
-): INodeExecutionData[] | null | undefined {
-	a.ok(runData[nodeName], `Can't find node with name '${nodeName}' in runData.`);
-	a.ok(
-		runData[nodeName][runIndex],
-		`Can't find a run for index '${runIndex}' for node name '${nodeName}'`,
-	);
-	a.ok(
-		runData[nodeName][runIndex].data,
-		`Can't find data for index '${runIndex}' for node name '${nodeName}'`,
-	);
-
-	return runData[nodeName][runIndex].data[connectionType][outputIndex];
+): INodeExecutionData[] | null {
+	return runData[nodeName]?.at(runIndex)?.data?.[connectionType].at(outputIndex) ?? null;
```

Comment:
I'd really like if we could do this for the rest of the code as well, instead of `[0][0][0]`. 👍🏽 
That said, do we want to keep these assertions, and report to sentry when data is not in the format we expect it to be? that could help improve the typings around this.

Reply:
I could add the assertions back, I removed them because I'm using the function in a different context now:

https://github.com/n8n-io/n8n/blob/pay-2167-executing-a-node-after-a-loop-node-does-not-work/packages/core/src/PartialExecutionUtils/findStartNodes.ts#L76-L93

Initially I wanted to make sure that I'm only accessing things that exist, but that part of the code now assumes that this may not exist. I could also try catch it there hand leave the assertions 🤔

Reply:
I'll merge it as is for now. Let me know what you think.